Sources/Stores/NetworkPersistableStore.swift

```diff
@@ -49,35 +50,41 @@ where Network.Remote == Data, Network.Request == URLRequest, Network.Response ==
 
         let cancelable = CancelableBag()
 
-        // 1st - Try to fetch from the Network
+        // try to fetch from the Network
         let networkCancelable = getNetworkPayload(resource) { [weak self] result in
 
             switch result {
-            case let .failure(error):
+            case .failure(.cancelled):
+                // The fetch was cancelled, so exit right away
+                completion(.failure(.cancelled))
+                return
+
+            case let .failure(networkError):
 
                 // Check if it's cancelled
                 guard cancelable.isCancelled == false else {
```

Comment:
Why is it necessary to check here if fetch is canceled? Shouldn't that case be caught on the `case .failure(.cancelled)`?

Reply:
Great question!

The `.cancelled` error that we receive here (from the `networkStack`) corresponds to having received an `NSError` with domain `NSURLErrorDomain` and code `NSURLErrorCancelled`.

This error can be caused not only any the user cancelling the `URLSessionTask` (in which case this check would be redundant), but also because of TLS/SSL issues for instance.

Hope this answers your question 😄

Reply:
Yep, it does 😄 
Thanks!